antarest/common/jwt.py

```diff
@@ -0,0 +1,63 @@
+from enum import Enum
+from typing import List, Union
+
+from dataclasses import dataclass, field
+
+from antarest.common.custom_types import JSON
+from antarest.common.roles import RoleType
+from antarest.login.model import Group, User
+
+
+@dataclass
+class JWTGroup:
+    id: str
+    name: str
+    role: RoleType
+
+    @staticmethod
+    def from_dict(data: JSON) -> "JWTGroup":
+        return JWTGroup(
+            id=data["id"],
+            name=data["name"],
+            role=RoleType.from_dict(data["role"]),
+        )
+
+    def to_dict(self) -> JSON:
+        return {"id": self.id, "name": self.name, "role": self.role.to_dict()}
+
+
+@dataclass
+class JWTUser:
+    id: int
+    name: str = ""
+    groups: List[JWTGroup] = field(default_factory=lambda: list())
+
+    @staticmethod
+    def from_dict(data: JSON) -> "JWTUser":
+        return JWTUser(
+            id=data["id"],
+            name=data["name"],
+            groups=[JWTGroup.from_dict(g) for g in data["groups"]],
+        )
+
+    def to_dict(self) -> JSON:
+        return {
+            "id": self.id,
+            "name": self.name,
+            "groups": [g.to_dict() for g in self.groups],
+        }
+
+    def is_site_admin(self) -> bool:
+        return "admin" in [g.id for g in self.groups]
+
+    def is_group_admin(self, groups: Union[Group, List[Group]]) -> bool:
```

Comment:
what is the use of this method when the arg is a list of group ? we won't know which group the user is admin of.

Reply:
currently is used only for get_user. A group admin should access to user only if group admin belongs to user group.  So we fetch user groups and we check that if user logged as an admin belongs to one of these groups.

Reply:
i see. maybe we should use any(is_group_admin(g) for g in groups) in the get_user method.
And is_group_admin should only take a single group.
Otherwise the result is ambiguous since we won't know if the user is admin of all groups or a single one which we won't know which.